index.bs

```diff
@@ -719,36 +974,48 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 		<h4 method for="Clipboard">writeText(|data|)</h4>
 		The {{Clipboard/writeText(data)}} method must run these steps:
 
-			1. Let |p| be a new [=Promise=].
+			1. Let |realm| be [=this=]'s [=relevant realm=].
+
+			1. Let |p| be [=a new promise=] in |realm|.
 
 			1. Run the following steps [=in parallel=]:
 
-				1. Let |r| be the result of running [=check clipboard write permission=] [=in parallel=]
+				1. Let |r| be the result of running [=check clipboard write permission=].
+
+				Note: Clipboard permission is not supported on Safari. However, the writeText() method must be called inside
+				a user gesture event and the user must select the paste option from the native context menu that pops up
+				when writeText() is called from JS, otherwise, the promise will be rejected.
 
-				1. If |r| is not "granted", then reject |p| with a "NotAllowedError" DOMException
+				1. If |r| is false, then:
+				
+					1. [=Queue a global task=] on the [=permission task source=], given |realm|'s [=Realm/global object=], to [=reject=] |p| with {{"NotAllowedError"}} DOMException in |realm|.
 
-				1. Let |newItemList| be an empty sequence&lt;{{ClipboardItem}}>.
+					1. Abort these steps.
 
-				1. Let |textBlob| be a new {{Blob}} created with:
-					[=type=] attribute set to <em>text/plain;charset=utf-8</em>, and
-					[=blobParts=] set to a sequence containing the <em>string</em> |data|.
+				1. [=Queue a global task=] on the [=clipboard task source=], given |realm|'s [=Realm/global object=], to perform the below steps:
 
-				1. Let |newItem| be a new {{ClipboardItem}} created with a single
-					representation:
-					[=type=] attribute set to <em>text/plain8</em>, and
-					[=data=] set to |textBlob|.
+					1. Let |newItemList| be empty [=clipboard items=].
```

Comment:
```suggestion
					1. Let |newItemList| be empty [=list=] of [=clipboard items=].
```

(might need to be `[=/list=]`, not sure.)

Reply:
Added definition of `clipboard items` and used that instead.